src/translate.py

```diff
@@ -3032,11 +3033,17 @@ def assign_phis(used_phis: List[PhiExpr], stack_info: StackInfo) -> None:
     while i < len(used_phis):
         phi = used_phis[i]
         assert phi.num_usages > 0
-        assert len(phi.node.parents) >= 2
+        # assert len(phi.node.parents) >= 2
+        if len(phi.node.parents) < 2:
+            i += 1
+            continue
         exprs = []
         for node in phi.node.parents:
             block_info = node.block.block_info
-            assert isinstance(block_info, BlockInfo)
+            # assert isinstance(block_info, BlockInfo)
+            if not block_info:
+                i += 1
+                continue
```

Comment:
This `continue`s the `for node in phi.node.parents`, but the `i += 1` makes it look like you want to continue the `while i < len(used_phis)` loop? 

Is this needed? I'd be kind of worried if we have non-terminal blocks without `block_info` -- I think even with re-writing, `block_info` can be preserved?

(also remove commented-out asserts)

Reply:
Yeah, the `block_info` issues are top on my list to debug. I'm not really sure why there are nodes that are lacking `block_info`...

Reply:
I put back the two `assert isinstance(..., BlockInfo)` checks + the `ValueError` except in `flow_graph.py` for `incoming_edges`, then re-ran mips2c against all of OOT, MM, & PM -- and there was no difference. Maybe some other refactor means that those cases no longer come up? (Seems like a good thing!)